pkg/orchestrator/scheduler/batch_service_job.go

```diff
@@ -92,13 +94,30 @@ func (b *BatchServiceJobScheduler) Process(ctx context.Context, evaluation *mode
 	execsByApprovalStatus.toApprove.markApproved(plan)
 	execsByApprovalStatus.toReject.markStopped(execRejected, plan)
 
+	// How many executions failed due to compute nodes rejecting bids?
+	rejected := existingExecs.filterRejected()
+
+	// If we have rejected executions, we need to retry in case the nodes accept
+	// them in future (eg, due to transient capacity issues). b.handleRetry()
+	// will do this for us, but it results in this same scheduler code being
+	// re-executed after the delay. Therefore, we must check
+	// evaluation.TriggeredBy to see if we are being triggered by a deferred
+	// retry; and if so, we do NOT just defer a retry into the future, as
+	// otherwise we'd loop forever.
+	if (len(rejected) > 0) && evaluation.TriggeredBy != models.EvalTriggerDefer {
+		b.handleRetry(plan, &job, len(rejected))
+		return b.planner.Process(ctx, plan)
+	}
+
 	// create new executions if needed
 	remainingExecutionCount := desiredRemainingCount - execsByApprovalStatus.activeCount()
 	if remainingExecutionCount > 0 {
-		allFailed := existingExecs.filterFailed().union(lost)
+		allFailed := existingExecs.filterFailed().union(lost).union(rejected)
 		var placementErr error
-		if len(allFailed) > 0 && !b.retryStrategy.ShouldRetry(ctx, orchestrator.RetryRequest{JobID: job.ID}) {
+		if len(allFailed) > 0 && !b.retryStrategy.ShouldRetry(ctx, orchestrator.RetryRequest{Job: &job}) {
 			placementErr = fmt.Errorf("exceeded max retries for job %s", job.ID)
```

Comment:
the retryStrategy is mainly invoked for failed executions to evaluate if the execution should retried on other nodes, or if the job is bad and we shouldn't retry any longer. `lost` are included here as well as the job might be the cause of taking the node down. Rejection on the other hand is different as the node rejects a job without even executing it, such as if the required engines are not supported, doesn't fit the total capacity, or no capacity currently available